### ISSUE TITLE
Enable CodeQL scanning ⚠️ 

### ISSUE THREAD

**Issue by JamieSlome** [finos/git-proxy#401]
```[tasklist]
### Tasks
- [x] Enable CodeQL (Extended)
- [x] Ensure CodeQL issues are resolved before merging into `main`
```
I'd like us to have a baseline tool in place and we can have future discussions about rigidifying the PR review process. @maoo, I have access to some of the repository settings but am not able to enable CodeQL. I believe this is only possible at the organization level or via administrative controls.

Thank you! 🎉

**Comment by JamieSlome** [comment]
I just remembered that this can also be enabled via the creation of a configuration file too 👍 The administrative "switch" is easier, but creating the config file will work too...

**Comment by JamieSlome** [comment]
@maoo - I've opened up #403 but will need you to "switch-on" the requirement for passing status check before PRs can be merged into `main`.

**Comment by maoo** [comment]
@JamieSlome - I've enabled `CodeQL analysis` , I see the CodeQL check that passed on #403 - please let me know if there's any tweaking to make.

**Comment by JamieSlome** [comment]
@maoo - thank you! Can you make the status check `required`, i.e. it has to pass before the merge can occur?

**Comment by maoo** [comment]
Done; since I was there, I made few more additions that IMO made a lot of sense; please let me know if I jumped the gun, I'm happy to revert the changes 😄 
- `Require conversation resolution before merging`
- `Require branches to be up to date before merging`
- `Status checks that are required` - `CodeQL` (new) and `EasyCLA`

**Comment by JamieSlome** [comment]
Thanks! The question is, why is https://github.com/finos/git-proxy/pull/403 now failing? 🤔

**Comment by maoo** [comment]
> Thanks! The question is, why is #403 now failing? 🤔

https://www.returngis.net/2023/05/como-solucionar-el-error-codeql-analyses-from-advanced-configurations-cannot-be-processed-when-the-default-setup-is-enabled-en-ghas/

Seems that specifying the CodeQL configuration via a GitHub Action addresses the issue; I'll raise a PR

**Comment by maoo** [comment]
https://github.com/finos/git-proxy/pull/406 - green! Feel free to review and merge

**Comment by JamieSlome** [comment]
Addressed in #406.

@maoo ❤️
